sherpa/models/model.py

```diff
@@ -1070,19 +1083,30 @@ class ArithmeticConstantModel(Model):
         is a scalar, otherwise it indicates an array of elements.
     """
 
-    def __init__(self, val, name=None):
-        val = SherpaFloat(val)
+    def __init__(self,
+                 # Use SupportsIndex rather than Sequence[SupportsFloat] to
+                 # avoid mypy warnings.
+                 val: Union[SupportsFloat, SupportsIndex],
+                 name: Optional[str] = None) -> None:
+
+        store = SherpaFloat(val)
         if name is None:
-            if numpy.isscalar(val):
-                name = str(val)
+            if np.isscalar(store):
+                name = str(store)
             else:
-                nstr = ','.join([str(s) for s in val.shape])
-                name = f'{val.dtype.name}[{nstr}]'
+                if TYPE_CHECKING:
+                    assert isinstance(val, np.ndarray)
+
+                # For some reason mypy didn't like
+                # '[str(s) for s in store.shape]`
+                dims = map(str, store.shape)
+                nstr = ','.join(dims)
+                name = f'{store.dtype.name}[{nstr}]'
```

Comment:
This is complicated, just to make a type checker happy - and that in a dynamically typed language. But fine.

Reply:
No, this was not my favourite bit of code...